Compiler error when using map lookup with "or {none}"

### Describe the bug

The following line 
```V
println(m[`4`] or {none})
```
Produces a compiler bug message

### Expected Behavior

The file compiles and prints "none"

### Current Behavior

Does not compile but fails with the following message:
```
/tmp/v_1000/maps.14566746918291051433.tmp.c:12306: error: identifier expected
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.
```

### Reproduction Steps

Create the a .v file with the following content and try to run or compile it:
```V
module main

fn main(){
	m := {
		`л`:1,
		`ő`:25,
	}
	println(m[`4`] or {none})
}
```

### Possible Solution

Check for "none" when checking for the type. Other types produce a proper compiler error message about not matching types

### Additional Information/Context

_No response_

### V version

0.3.3 15cb18c

### Environment details (OS name and version, etc.)

OS: linux, Linux Mint 20.3
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/laszlo/$TMP/2023-03/vtest
vmodules: /home/laszlo/.vmodules
vroot: /var/tmp/vlang
vexe: /var/tmp/vlang/v
vexe mtime: 2023-02-27 09:37:33
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 15cb18c

Git version: git version 2.25.1
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1160,7 +1160,7 @@ fn (mut c Checker) check_or_last_stmt(stmt ast.Stmt, ret_type ast.Type, expr_ret
 						&& c.table.sym(last_stmt_typ).kind == .voidptr {
 						return
 					}
-					if last_stmt_typ == ast.none_type_idx {
+					if last_stmt_typ == ast.none_type_idx && ret_type.has_flag(.option) {
 						return
 					}
 					type_name := c.table.type_to_str(last_stmt_typ)

--- a/vlib/v/checker/tests/wrong_none_on_or_block_err.out
+++ b/vlib/v/checker/tests/wrong_none_on_or_block_err.out
@@ -1,0 +1,6 @@
+vlib/v/checker/tests/wrong_none_on_or_block_err.vv:8:22: error: wrong return type `none` in the `or {}` block, expected `int`
+    6 |         `ő`: 25
+    7 |     }
+    8 |     println(m[`4`] or { none })
+      |                         ~~~~
+    9 | }

--- a/vlib/v/checker/tests/wrong_none_on_or_block_err.vv
+++ b/vlib/v/checker/tests/wrong_none_on_or_block_err.vv
@@ -1,0 +1,9 @@
+module main
+
+fn main() {
+	m := {
+		`л`: 1
+		`ő`: 25
+	}
+	println(m[`4`] or { none })
+}

